front/temporal/usage_queue/activities.ts

```diff
@@ -27,8 +27,10 @@ export async function recordUsageActivity(workspaceId: string) {
   logger.info({}, "[UsageQueue] Recording usage for worskpace.");
 
   if (!subscription) {
```

Comment:
🤔 In this current approach, we do not verify the subscription before creating the workflow, and instead, we defer all the logic to this point. Initially, this check was implemented to ensure that all active workspaces have a valid subscription. I would push for keeping this error, but relax it a bit. By looking at the previous subscription, we can confirm if the end date is recent enough to disregard the error safely.

Reply:
That's not the purpose of that workflow is it?

Reply:
In this design, the aim was to prevent fetching the subscription before initiating the workflow, treating the workflow as a queue system under the assumption that all active workspaces have a subscription. However, this implementation did not consider the scenario where a workspace is churned with an event already queued.

To address this, I can either:
* Relax the error check and ignore those recently churned workspaces
* Modify the function prototype to start the usage workflow to rely on the `auth` object, which is nearly available at all call sites and assert that there is a valid subscription so we only enqueue event for workspaces with a subscription and we can keep your code.

Reply:
Isn’t 1 what we’re doing here?

Anything non timing based win

Reply:
Not exactly, 1. ignores all workspaces without any subscriptions no matter if they are recently churned or are enterprise but are missing a valid subscription. Anyway, will keep like this for now and monitor the logs.